hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/util/Util.java

```diff
@@ -301,4 +309,29 @@ public static MonitorServiceConfiguration getMonitorServiceConfiguration(Operati
         MonitorServiceConfiguration config = new MonitorServiceConfigurationBuilder(subsystemConfig, context).build();
         return config;
     }
+
+    /**
+     * Tries to determine the system ID for the machine where this JVM is located.
+     *
+     * @return system ID or null if cannot determine
+     */
+    public static String getSystemId() {
+        if (systemId == null) {
+            File machineIdFile = new File("/etc/machine-id");
+            if (machineIdFile.exists() && machineIdFile.canRead()) {
+                try (Reader reader = new InputStreamReader(new FileInputStream(machineIdFile))) {
+                    systemId = new BufferedReader(reader).lines().collect(Collectors.joining("\n"));
+                } catch (IOException e) {
+                    log.errorf(e, "/etc/machine-id exists and is readable, but exception was raised when reading it");
+                    systemId = "";
```

Comment:
if we keep it at null, it will keep attempting to read the file (and keep getting exceptions) everytime getSystemId() is called. This just forces systemId to be cached with a non-null value so we don't continue to try to obtain it when its just going to keep failing.